src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -1267,6 +1267,9 @@ def build_vmss_resource(cmd, name, computer_name_prefix, location, tags, overpro
         vmss_properties['upgradePolicy'] = {
             'mode': upgrade_policy_mode
         }
+        if orchestration_mode.lower() == 'flexible':
+            vmss_properties['upgradePolicy'] = None
```

Comment:
Please add comments to explain why we need this logic

Reply:
> (NEW) null or [] for upgrade policy instead of 'manual'

Reply:
```suggestion
        if orchestration_mode.lower() == 'flexible':
            vmss_properties['upgradePolicy'] = None
            upgrade_policy_mode = None
```

Reply:
> what is the current value/default for "upgradePolicy" when customer makes a VMSS Flexible deployment on CLI today? If it is not already null/manual, we should set upgradePolicy to null with this update.

Reply:
https://github.com/Azure/azure-cli/blob/3abbcbb243c8904f7a71b199b99521e66e493a3d/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L1676-L1684

It has been set to None here, so these two lines are not needed